.circleci/config.in.yml

```diff
@@ -73,6 +90,37 @@ jobs:
             python .circleci/regenerate.py
             git diff --exit-code || (echo ".circleci/config.yml not in sync with config.yml.in! Run .circleci/regenerate.py to update config"; exit 1)
 
+  test_promote_s3:
+    <<: *promote_common
+    steps:
+      - checkout
+      - run:
+          name: Setting test environment variables
+          command: |
+            export PYTORCH_S3_FROM=nightly
```

Comment:
Does this actually work? I find it hard to believe that CircleCI is propagating environment variables across steps. Note that for example `binary_populate_env.sh` works by building a script which the final docker environment sources to set a bunch of envvars.